hts_internal.h

```diff
@@ -120,6 +120,22 @@ int bgzf_idx_push(BGZF *fp, hts_idx_t *hidx, int tid, hts_pos_t beg, hts_pos_t e
  */
 void bgzf_idx_amend_last(BGZF *fp, hts_idx_t *hidx, uint64_t offset);
 
+static inline int find_file_extension(const char *fn, char ext_out[5])
```

Comment:
The array size here is incorrect.  It doesn't strictly matter because C treats `ext_out` as a `char *` and the array passed by the caller is big enough.  It is misleading though so should be corrected.  We could also use a C99 feature and turn it into `ext_out[static 7]` which means it's treated more like you'd expect (a `char` array with at least 7 elements).  `gcc` seems to ignore the declared size here, but `clang` will complain if it's bigger than the array passed in by the caller.